vktrace/vktrace_replay/vkreplay_vkreplay.cpp

```diff
@@ -4462,28 +4463,26 @@ VkResult vkReplay::manually_replay_vkGetDisplayPlaneSupportedDisplaysKHR(packet_
 
     // No need to remap planeIndex
     // No need to remap pDisplayCount
+    // No need to remap pDisplays
 
-    // Get remapped displays
-    VkDisplayKHR *remapped_displays = VKTRACE_NEW_ARRAY(VkDisplayKHR, *pPacket->pDisplayCount);
-    for (uint32_t i = 0; i < *pPacket->pDisplayCount; ++i) {
-        remapped_displays[i] = m_objMapper.remap_displaykhrs(*(pPacket->pDisplays + i));
+    VkDisplayKHR *pDisplays = pPacket->pDisplays;
+    if (pPacket->pDisplays != NULL) {
+        pDisplays = VKTRACE_NEW_ARRAY(VkDisplayKHR, *pPacket->pDisplayCount);
     }
+    replayResult = m_vkFuncs.GetDisplayPlaneSupportedDisplaysKHR(remappedphysicalDevice, pPacket->planeIndex,
+                                                                 pPacket->pDisplayCount, pDisplays);
 
-    if (pPacket->pDisplays != VK_NULL_HANDLE && remapped_displays == VK_NULL_HANDLE) {
-        vktrace_LogError("Error detected in GetDisplayPlaneSupportedDisplaysKHR() due to invalid remapped VkDisplayKHR.");
-        return VK_ERROR_VALIDATION_FAILED_EXT;
-    }
-
-    VkDisplayKHR *displays = pPacket->pDisplays;
-
-    auto result = m_vkFuncs.GetDisplayPlaneSupportedDisplaysKHR(remappedphysicalDevice, pPacket->planeIndex, pPacket->pDisplayCount,
-                                                                remapped_displays);
-
-    if (memcmp(displays, pPacket->pDisplays, sizeof(VkDisplayKHR) * (*pPacket->pDisplayCount)) != 0) {
-        vktrace_LogError("Display Plane supported displays differ. Displays may not match as expected.");
+    if (pPacket->pDisplays != NULL) {
+        if (memcmp(pDisplays, pPacket->pDisplays, sizeof(VkDisplayKHR) * (*pPacket->pDisplayCount)) != 0) {
```

Comment:
I don't understand why this memcmp should be done. The values in pDisplays will be values from the replay device, the values in pPacket->pDisplays will be from the trace device. It seems an error would be generated a vast majority of the time when this code is executed. Have you seen this code execute without generating an error?

The old code also did the compare, but it looks like it was comparing the values returned to themselves.

Reply:
This piece of code is inherited from the old code.  Yes, you are correct, the memcmp should be removed since the values from display device is not expected to be the same as the values on trace device.